pkg/docker/containers-view.jsx

```diff
@@ -530,15 +530,19 @@ var ImageList = React.createClass({
     deleteImage: function (image, event) {
         if (event.button !== 0)
             return;
-
-        util.confirm(cockpit.format(_("Delete $0"), image.RepoTags[0]),
-                     _("Are you sure you want to delete this image?"), _("Delete"))
-                .done(function () {
-                    this.props.client.rmi(image.Id)
-                            .fail(function(ex) {
-                                util.show_unexpected_error(ex);
-                            });
+        util.delete_image_confirm(this.props.client, image).done(
+            function(runningContainers, force) {
+                var stopPromises = [];
+                if (runningContainers.length > 0)
+                    $(runningContainers).each(function(index, value) {
```

Comment:
Can we please avoid introducing new jQuery code? Maybe something like this?

    var stopPromises = runningContainers.map(id => this.props.client.stop(id));

which is more succinct and avoids jQuery?

This is already JSX, ergo ES6, so you can use arrow functions, and avoid the silly `.bind(this)` thing, too.